tensorflow-tools/src/main/java/org/tensorflow/tools/buffer/DoubleDataBuffer.java

```diff
@@ -157,4 +157,9 @@ default DoubleDataBuffer narrow(long size) {
 
   @Override
   DoubleDataBuffer slice(long index, long size);
+
+  @Override
+  default DataBufferWindow<DoubleDataBuffer> window(long size) {
+    return null;
```

Comment:
What's with all the default methods returning null? I didn't see null checks elsewhere in the code to deal with it not being overridden?

Reply:
[here](https://github.com/tensorflow/java/blob/442d65d546db4704422e6f98c7b9c4172fecabef/tensorflow-tools/src/main/java/org/tensorflow/tools/ndarray/impl/dense/AbstractDenseNdArray.java#L45)

I just implemented windows for raw buffers at the moment. At some point, they should have a window implementation but it is not mandatory.

Unfortunately in Java, when you override the signature of a method in a super interface, you don’t inherit anymore of its default implementation, so I had to return null explicitly in all subinterfaces.

Reply:
Ok, in which case the javadoc should probably note that the window method is optional on `DataBuffer`, similar to the remove methods on `List`. And maybe it should throw `UnsupportedOperationException`? It would be kinda ugly to have the method that uses it catch a RuntimeException though so maybe returning null is fine.

Reply:
Yeah, I was hesitating between the two... but probably after this PR I'll add windows to all other buffer types as well, so it will really be an exception if it is not supported (e.g. if the user provides one of its own buffer implementation). So maybe the exception is better, I'll change it.